### PR TITLE
Add documentation on how to programmatically access a selection data.

### PR DESCRIPTION
**Pull request by senotrusov**
This pull request includes a
- [x] Documentation

The following changes were made
- I added the section in documentation

If this is related to an existing ticket, include a link to it as well.
#4334

### CODE REVIEW

**Comment by senotrusov** [comment]
In my understanding, that failed CI build have nothing to do with my changes.

File: docs/_includes/options/data/selection-access.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+$('select').select2('data');
+{% endhighlight %}
+
+  <p>
+    As Select2 uses the HTML <code>&lt;SELECT&gt;</code> element to store the selection result, the selection data are represented by <code>&lt;OPTION&gt;</code> elements and can be accessed in the plain jQuery/DOM manner. The resulting elements will have properties/values from the source data objects, stored by the means of jQuery <code>data()</code> method and accessible by key <code>'data'</code>:
+  </p>
+
+{% highlight js linenos %}
+// Retrieve source data object's data of the first selected element
+$('select').find(':selected').data('data');
```
Comment (kevin-brown) [inline docs/_includes/options/data/selection-access.html]: The locally cached `data` key on the `<option>` is an implementation detail that is definitely private API. It's actually something we're planning on moving away from in the future.

**Comment by senotrusov** [comment]
@kevin-brown Based on your comment, I removed the mentions of locally cached `data` key on the `<option>`. Thank you!

**Comment by subssn21** [comment]
Please merge this documentation or rewrite it somewhere else in the documentation. I just spent an hour trying to figure out how to access the data, and only found the answer by looking through the pull requests and happening across this.

**Comment by jpic** [comment]
Awesome, we like more documentation !

**Comment by senotrusov** [comment]
@jpic Awesome, thanks a lot!

**Pull request merged**